src/handlers/psp.ts

```diff
@@ -166,32 +165,65 @@ const initializeState = (config: Config): State => {
   };
 };
 
+const wait = (ms: number) => new Promise((res) => setTimeout(res, ms));
+const doRequestWithRetries = async (
+  fn: () => Promise<CallApiResult>,
+  minDelay = 1_000,
+  maxDelay = 5_000
+): Promise<CallApiResult> => {
+  try {
+    return await fn();
+  } catch (e) {
+    await wait(Math.floor(Math.random() * (maxDelay - minDelay + 1) + minDelay));
+    return await doRequestWithRetries(fn, minDelay, maxDelay);
+  }
+};
+
 const executeApiCalls = async (state: State): Promise<State> => {
   const { config, baseLogOptions, groupedSubscriptions } = state;
 
-  const apiValuePromises = groupedSubscriptions.map(({ subscriptions, template, endpoint }) => {
+  let hasApiCallWrapperTimedout = false;
+  const responses: CallApiResult[] = [];
+  const apiValuePromises = groupedSubscriptions.map(async ({ subscriptions, template, endpoint }) => {
     const apiCallParameters = abi.decode(template.templateParameters);
-    return go(
-      async () => {
-        const [logs, data] = await callApi(config, endpoint, apiCallParameters);
-        return [logs, { templateId: template.id, apiValue: data, subscriptions }] as node.LogsData<{
-          templateId: string;
-          apiValue: ethers.BigNumber | null;
-          subscriptions: Id<Subscription>[];
-        }>;
-      },
-      { timeoutMs: TIMEOUT_MS, retries: RETRIES }
-    );
+
+    let hasApiCallTimedout = false;
+    const result = await doRequestWithRetries(async () => {
+      if (hasApiCallWrapperTimedout) {
+        return [[], { templateId: template.id, apiValue: null, subscriptions }] as CallApiResult;
+      }
+      const result = await go(
+        async () => {
+          try {
+            return await callApi(config, endpoint, apiCallParameters);
+          } catch (err) {
+            if (hasApiCallTimedout) {
+              return [
+                [utils.logger.pend('ERROR', '' + err)],
+                { templateId: template.id, apiValue: null, subscriptions },
+              ] as CallApiResult;
+            }
```

Comment:
I haven't been able to get rid of this flag because of an edge case where the callApi() call has timed out (more that 10sec) and the callApi() throws an error after this time out. So I use this flag to handle any error that might occur after timing out.

Reply:
@amarthadan showed me an example that, but the issue is actually inside the `go` function. 

Specifically, it's because of a bug inside the `attemp` library here https://github.com/lifeomic/attempt/blob/master/src/index.ts#L187 where it calls on error which throws here https://github.com/lifeomic/attempt/blob/master/src/index.ts#L146

This is problematic, because it breaks the promise chain and when you do `Promise.resolve(somePromise)` it returns `somePromise` which is not handled there. The fix would be to add a `catch` clause after it. But honestly, promises should not be used like this, you should strive to make linear promise chains which are easy to reason about, which is not the case of that library. I'll take care of `promise-utils` and let Burak publish it. 

As of now, I think having something that works is fine. I'd still prefer [this implementation](https://github.com/api3dao/airkeeper/commit/ff26f2f872e00a38323ecdff8326cd675928b4f2), but I'd just create a JIRA issue and we can update it after there is a new version of go utils. WDYT?